app/webapp/src/main/java/uk/ac/ebi/fg/annotare2/web/gwt/editor/client/view/experiment/design/strategy/FileTypeMappingStrategyFactory.java

```diff
@@ -0,0 +1,17 @@
+package uk.ac.ebi.fg.annotare2.web.gwt.editor.client.view.experiment.design.strategy;
+
+import uk.ac.ebi.fg.annotare2.submission.model.ExperimentProfileType;
+
+public class FileTypeMappingStrategyFactory {
+    public static FileTypeMappingStrategy getStrategy(ExperimentProfileType experimentType){
+        if(experimentType.isSequencing()){
```

Comment:
I would like to remove this not nice conditional branching.
I was hoping that you can do an `abstract` class and subclasses.
Or you can do method overloading in this class...
What do you think?

Reply:
Added a Hashmap to hold strategy supplier for all types. Replaced conditional branching and getting strategy instances from the map.